java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/SuffixedPartition.java

```diff
@@ -0,0 +1,28 @@
+package com.godaddy.asherah.appencryption;
+
+public class SuffixedPartition extends Partition {
+
+  private final String regionSuffix;
+
+  public SuffixedPartition(final String partitionId, final String serviceId, final String productId,
+                           final String regionSuffix) {
+    super(partitionId, serviceId, productId);
+    this.regionSuffix = regionSuffix;
+  }
+
+  @Override
+  public String getSystemKeyId() {
+    return "_SK_" + getServiceId() + "_" + getProductId() + "_" + this.regionSuffix;
+  }
+
+  @Override
+  public String getIntermediateKeyId() {
+    return "_IK_" + getPartitionId() + "_" + getServiceId() + "_" + getProductId() + "_" + this.regionSuffix;
+  }
+
+  @Override
+  public String toString() {
+    return getClass().getSimpleName() + "[partitionId=" + getPartitionId() +
+      ", serviceId=" + getServiceId() + ", productId=" + getProductId() + ", regionSuffix=" + this.regionSuffix + "]";
+  }
+}
```

Comment:
one approach you can take is having the default implementations in the abstract class `Partition`.

The `DefaultPartition` class would then just reduce to a constructor with default implementations.
The `SuffixedPartition` class would then call the default implementations and just add ```"_" + this.regionSuffix;```

Reply:
Personally, I like having separate implementations better. It keeps things simpler. Any  particular advantage that you foresee using the above approach?

Reply:
this has been taken care of